### ISSUE TITLE
save preferences in searching 

### ISSUE THREAD

**Issue by lonkle** [NodeBB/NodeBB#8784]
- **NodeBB version:** 
- **NodeBB git hash:** 

- **Installed NodeBB Plugins:** 

- **Database type:** mongo, redis, or postgres
- **Database version:** 

- **Exact steps to cause this issue:**

- **What you expected:**

- **What happened instead:**

**Comment by barisusakli** [comment]
Can you provide reproduction steps?

**Comment by lonkle** [comment]
@barisusakli 

1: Go to: https://nodebbinstallation.com/search right next to the search input field, you'll see an `In` dropdown box to specify if you want to specify just `Title` or the default `Title and Posts`. 
2: Select the "Title" from the dropdown box.
3: Select "Advanced Search" and add anyone to the `Posted by` multi-select input field.
4: Click "Save preferences"
5: Reload "https://nodebbinstallation.com/search" and all of your preferences will be correctly saved and you'll see all of your preferences pre-loaded for you **as well as** results. 
6: Unfortunately, the DB query and shown post results **only** seem to take into account the singular search term and none of the other saved search preferences.
7: If you hit the "Search" button after reloading the Search page, then it does takes into account **all** of the saved preferences and the DB query and results are correct based on the Saved Search Preferences. But I'd assume users would expect them to be correct on initial page load.

Did that make sense?

**Comment by barisusakli** [comment]
Yeah this happens because the search preferences are saved into local storage client side so when you reload the page on `/search` server has no idea of your preferences.

The preferences are used when you click search on the advanced search page or when you make a quick search in the header. 

I think a simple fix would be to refresh the page with the correct url parameters when you click save preferences.

**Comment by lonkle** [comment]
Ah, I understand, so the preferences are changed via Javascript very quickly after loading the page. And the results are just "default" results.

I'll just make a quick TamperMonkey JS script to click the "Search" button for me whenever I load `/search` so whenever I go to the `/search` URL, the results will be what I expect them to be by default from the Saved Preferences. Should be one or two line script.

Thank you for clarifying how it works so I could find a workaround! ☺️

**Comment by lonkle** [comment]
I do want to note that saving this server side, would be helpful for users and the efficiency of your app. Not only would it allow for instantly querying the exact saved preference (which does cut down on DB queries), but also users then wouldn't have to add our saved preferences to each device individually. Or re-add them when we erase data in our browsers or use new ones or use incognito windows. Just seems an odd choice to save specifically that data within the browser with all the cons in doing so IMO.

**Comment by mySebbe** [comment]
Implemented in #14665. Saving preferences now runs the existing advanced-search path with the current form values, so the filters are applied immediately and included in the reloadable URL.
